src/compiler/checker.ts

```diff
@@ -5750,6 +5755,25 @@ namespace ts {
                 }
                 return Ternary.False;
             }
+
+            function enumRelatedTo(source: Type, target: Type) {
+                if (source.symbol.name !== target.symbol.name) {
+                    return Ternary.False;
+                }
+                const sourceDecl = <EnumDeclaration>getMergedSymbol(source.symbol).valueDeclaration;
+                const targetDecl = <EnumDeclaration>getMergedSymbol(target.symbol).valueDeclaration;
+                const targetMembers = arrayToMap(targetDecl.members, member => getTextOfPropertyName(<PropertyName>member.name));

```

Comment:
You never use `targetMembers`, so there's no need to initialize it like this. I'd recommend just passing in `x => x` or `x => true`